examples/android_app/app/src/main/java/com/example/llamastackandroiddemo/LlamaStackRemoteInference.kt

```diff
@@ -0,0 +1,167 @@
+package com.example.llamastackandroiddemo
+
+import android.content.Context
+import com.llama.llamastack.client.LlamaStackClientClient
+import com.llama.llamastack.client.okhttp.LlamaStackClientOkHttpClient
+import com.llama.llamastack.models.CompletionMessage
+import com.llama.llamastack.models.InferenceChatCompletionParams
+import com.llama.llamastack.models.SamplingParams
+import com.llama.llamastack.models.SystemMessage
+import com.llama.llamastack.models.UserMessage
+import java.time.Instant
+import java.time.ZoneId
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+import java.util.concurrent.CompletableFuture
+import kotlinx.datetime.Clock
+
+
+class LlamaStackRemoteInference(remoteURL: String) {
+
+    private var client: LlamaStackClientClient? = null
+
+    init {
+        try {
+            client = LlamaStackClientOkHttpClient
+                .builder()
+                .baseUrl(remoteURL)
+                .build()
+        } catch (e: Exception) {
+            client = null
+            AppLogging.getInstance().log(e.message)
+        }
+    }
+
+    fun inferenceStart(modelName: String, temperature: Double, prompt: ArrayList<Message>, systemPrompt:String, ctx: Context): String {
+        val future = CompletableFuture<String>()
+        //Get the current time in ISO format and pass it to the model in system prompt as a reference. This is useful for any scheduling and vague timing reference from user prompt.
+        val zdt = ZonedDateTime.ofInstant(Instant.parse(Clock.System.now().toString()), ZoneId.systemDefault())
+        val formattedZdt = zdt.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"))
+        val availableFunctions = AvailableFunctions.getInstance()
+        val functionDefinitions = availableFunctions.values()
+        var sysPrompt = systemPrompt
+        if (sysPrompt == "") {
```

Comment:
As discussed, this definitely needs platform-ized and belongs in the client SDK using some sort of Handlebars-esque templating library. We can tackle next week.

Reply:
Make sense